src/app/seller-post-ad-form/seller-post-ad-form.component.ts

```diff
@@ -0,0 +1,181 @@
+    this.imageArray[id].heroImage = "none";
+    this.imageArray[id].imageLoaderProperty="";
+  }
+
+  async delay(ms: number,id) {
+    await new Promise(resolve => setTimeout(()=>resolve(), ms)).then(()=>console.log("image Loading"));
+    this.imageArray[id].addEditProperty="";
+    this.imageArray[id].CrossBtnValue="";
+    this.imageArray[id].imageDisplayValue="";
+    this.imageArray[id].imageURL = "https://images.pexels.com/photos/170811/pexels-photo-170811.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500";
+    this.imageArray[id].buttonName ="Change";
+    this.imageArray[id].iconOfButton = "edit";
+    this.imageArray[id].imageLoaderProperty="none";
+    if(id==0)
+    {
+      this.selectHeroImg(id);
+      this.imageArray[0].heroImage="";
+    }
+  }
+  
+  addImage(id,event){
+    if(id==0)
+    {
+      this.selectHeroImg(id);
+      this.imageArray[0].heroImage="";
+    }
+    
+    this.imageLoader(id);
+    this.delay(3000,id);
+    console.log(event.target.value);  // calls upload image api it will return url of image after uploading image
+    //this.imageArray[id].imageURL = funcUploadImage(event.target.value);
+    //this.imageArray[id].imageURL = "https://images.pexels.com/photos/170811/pexels-photo-170811.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500";
+    //Mock image URL
+
+    //this.imageArray[id].addEditProperty="";
+    // this.imageArray[id].CrossBtnValue="";
+    // this.imageArray[id].imageDisplayValue="";
+    // this.imageArray[id].buttonName ="Change";
+    // this.imageArray[id].iconOfButton = "edit";
+    // this.imageArray[id].imageLoaderProperty="none";
+
+    this.imageCounter += 1;
+    if(this.imageCounter <= this.maxNoOfImage)
+    {
+      let image =new ImageProperty();
+      this.imageArray.push(image);
+    }
+  }
+ 
+  removeImage(id)
+  {
+    if(this.imageCounter!=0 && this.imageArray[id].pictureContainerStyle =="4px solid blue")
+    {
+      this.selectHeroImg(0);
+      this.imageArray[0].heroImage="";
+    }
+
+    this.imageArray[id].iconOfButton = "plus";
+    this.imageArray[id].imageDisplayValue = "none";
+    this.imageArray[id].CrossBtnValue = "none";
+    this.imageArray[id].buttonName = "Add";
+    this.imageArray[id].iconOfButton = "plus";
+    this.imageArray[id].pictureContainerStyle = "1px solid lightgrey";
+
+    if(this.imageCounter>this.minNoOfImage)
+    {
+      this.imageArray.splice(id,1);
+      this.imageCounter -= 1;
+    }
+    if(this.imageCounter==this.maxNoOfImage)
+    {
+      let image =new ImageProperty();
+      this.imageArray.push(image);
+    }
+   
+  }
+
+  selectHeroImg(id)
+  {
+    for (let index = 0; index < this.imageArray.length; index++) 
+    {
+      if(index!=id)
+      {
+        this.imageArray[index].pictureContainerStyle = "1px solid lightgrey";
+        this.imageArray[index].heroImage="none";
+      }      
+    }
+    this.imageArray[id].pictureContainerStyle = "4px solid blue";
```

Comment:
Never write css in component.ts
Use class.
Handle this in html itself.
This logic should not be in component class.

Reply:
+1 on this. Rather than doing this, you can write the css for this id in the css class itself. If it finds the id in html it'll automatically apply this css